cpp/src/DataStorm/NodeI.cpp

```diff
@@ -458,6 +458,24 @@ NodeI::retrySubscriberSessionCreation(
     const shared_ptr<SubscriberSessionI>& session,
     exception_ptr ex)
 {
+    try
+    {
+        rethrow_exception(ex);
+    }
+    catch (const SessionCreationException& sessionCreationException)
+    {
+        if (sessionCreationException.error == SessionCreationError::AlreadyConnected && session->checkSession())
+        {
+            // A concurrent session attempt from the peer succeeded, no need to retry.
```

Comment:
In this case both peers are seeing the session as connected.

- This node Session::checkSession returns true, meaning it is already connected.
- The peer node just throw `SessionCreationException` with `AlreadyConnected` error code.

There is no need to keep retrying, and if we keep the retries will keep failing with already connected and the session will be destroyed when the retry limit is hit.